gamemodes/terrortown/entities/entities/sidekick/shared.lua

```diff
@@ -57,6 +57,52 @@ if CLIENT then
 	end)
 end
 
+function GetDarkenColor(color)
+	if not istable(color) then return end
+	local col = table.Copy(color)
+	-- darken color
+	for _, v in ipairs{"r", "g", "b"} do
+		col[v] = col[v] - 60
+		if col[v] < 0 then
+			col[v] = 0
+		end
+	end
+
+	col.a = 255
+
+	return col
+end
+
+local function tmpfnc(ply, mate, colorTable)
+	if IsValid(mate) and mate:IsPlayer() then
+		return table.Copy(mate:GetSubRoleData()[colorTable])
```

Comment:
This isn‘t equal with ply.GetRole(Dk/Bg)Color :/

Reply:
What do you mean exactly? I haven't touched the function at all, just moved it from client-only to shared for the TTT2ModifyRoleColor hook.

Reply:
yea but what about using the player functions?